src/main/java/com/uid2/shared/attest/JwtService.java

```diff
@@ -0,0 +1,142 @@
+package com.uid2.shared.attest;
+
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.Optional;
+
+import com.uid2.shared.Const;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import com.google.api.client.json.webtoken.JsonWebSignature;
+import com.google.api.client.json.webtoken.JsonWebToken;
+import com.google.auth.oauth2.TokenVerifier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JwtService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JwtService.class);
+    private final JsonObject config;
+
+    private final HashSet<PublicKey> publicKeys = new HashSet<>();
+
+
+    public JwtService(JsonObject config) {
+        this.config = config;
+        JsonArray publicKeys = config.getJsonArray(Const.Config.AwsKmsJwtSigningPublicKeysProp);
+        if (publicKeys == null || publicKeys.isEmpty()) {
+            LOGGER.error("Unable to read public keys from the configuration. JWTs can not be verified.");
+            return;

```

Comment:
Can this recover? Should we throw an exception instead of keeping `publicKeys` null and then erroring at the consumption?

Reply:
There will be cases where JWT validation is not required, so the hosting service can set the config to not to enforce JWT validation. They would then not want to provide public keys. Also, the NullPointer exception for validateJwt would be caught and returned as an invalid JWT, rather than thrown.

Reply:
Should we still log this as an error, then?

Reply:
I'll change it to a info, as it may not be an error in that case